codec.go

```diff
@@ -31,10 +31,22 @@ type Codec interface {
 	// WriteResponse must be safe for concurrent use by multiple goroutines.
 	WriteResponse(*Response, interface{}) error
 
+	IsV2() bool
```

Comment:
I believe adding this function to the Codec interface might not be the best way to proceed. The rpc package is designed to work independently of any specific codec. The current change, however, seems to bind it with json-rpc.

Reply:
Agreed. Problem is that I need to know whether to send an empty string as error or nil in case nil is returned from the client. 
But if it cannot be united in a sensible way, I will just clean up my repo to only have JSON-RPC 2.0.